airbyte-integrations/connectors/destination-bigquery-denormalized/src/test/java/io/airbyte/integrations/destination/bigquery/BigQueryDenormalizedDestinationTest.java

```diff
@@ -0,0 +1,424 @@
+    final Map<AirbyteStreamNameNamespacePair, AbstractBigQueryUploader<?>> uploaderMap = new HashMap<>();
+    final String streamName = "stream_name";
+    final String nameSpace = "name_space";
+    final Table tableMock = mock(Table.class);
+    final Schema schemaMock = mock(Schema.class);
+    final TableDefinition tableDefinitionMock = mock(TableDefinition.class);
+
+    when(uploaderConfigMock.getConfigStream()).thenReturn(configuredStreamMock);
+    when(uploaderConfigMock.getBigQuery()).thenReturn(bigQueryMock);
+    when(uploaderConfigMock.getFormatter()).thenReturn(bigQueryRecordFormatterMock);
+    when(configuredStreamMock.getStream()).thenReturn(airbyteStreamMock);
+    when(tableMock.getDefinition()).thenReturn(tableDefinitionMock);
+    when(uploaderConfigMock.getTargetTableName()).thenReturn("target_table");
+    when(airbyteStreamMock.getNamespace()).thenReturn(nameSpace);
+    when(airbyteStreamMock.getName()).thenReturn(streamName);
+    when(bigQueryMock.getTable(anyString(), anyString())).thenReturn(tableMock);
+
+    // expected schema is not null
+    when(bigQueryRecordFormatterMock.getBigQuerySchema()).thenReturn(schemaMock);
+    // existing schema is null
+    when(tableDefinitionMock.getSchema()).thenReturn(null);
+    // run test
+    bqdd.putStreamIntoUploaderMap(airbyteStreamMock, uploaderConfigMock, uploaderMap);
+    // should use LegacyArrayFormatter
+    verify(bigQueryRecordFormatterMock, times(1)).setArrayFormatter(any(LegacyArrayFormatter.class));
+  }
+
+  @Test
+  void putStreamIntoUploaderMap_compareSchemas_existingAndExpectedAreNull() throws IOException {
+    final Map<AirbyteStreamNameNamespacePair, AbstractBigQueryUploader<?>> uploaderMap = new HashMap<>();
+    final Table tableMock = mock(Table.class);
+    final TableDefinition tableDefinitionMock = mock(TableDefinition.class);
+
+    when(uploaderConfigMock.getConfigStream()).thenReturn(configuredStreamMock);
+    when(uploaderConfigMock.getBigQuery()).thenReturn(bigQueryMock);
+    when(uploaderConfigMock.getFormatter()).thenReturn(bigQueryRecordFormatterMock);
+    when(configuredStreamMock.getStream()).thenReturn(airbyteStreamMock);
+    when(uploaderConfigMock.getTargetTableName()).thenReturn("target_table");
+    when(airbyteStreamMock.getNamespace()).thenReturn("name_space");
+    when(airbyteStreamMock.getName()).thenReturn("stream_name");
+    when(bigQueryMock.getTable(anyString(), anyString())).thenReturn(tableMock);
+    when(tableMock.getDefinition()).thenReturn(tableDefinitionMock);
+
+    // expected schema is null
+    when(bigQueryRecordFormatterMock.getBigQuerySchema()).thenReturn(null);
+    // existing schema is null
+    when(tableDefinitionMock.getSchema()).thenReturn(null);
+    // run test
+    bqdd.putStreamIntoUploaderMap(airbyteStreamMock, uploaderConfigMock, uploaderMap);
+    // should not use LegacyArrayFormatter
+    verify(bigQueryRecordFormatterMock, times(0)).setArrayFormatter(any(LegacyArrayFormatter.class));
```

Comment:
what _should_ this use?

Reply:
the test is for `putStreamIntoUploaderMap` method functionality so it check the logic of this method. In this logic we are setting 
`((DefaultBigQueryDenormalizedRecordFormatter) formatter).setArrayFormatter(new LegacyArrayFormatter());`
or no. 
We can check if it use the default formatter instead but then I suppose it will be similar to integration test.

Reply:
ah yeah, that makes sense. I think this is fine as-is.